client/src/pages/user/UserProfile.tsx

```diff
@@ -0,0 +1,205 @@
+  });
+
+  const [paginationProps, setPaginationProps] = useState<PaginationRequest>({ pageNumber: 0, pageSize: 5 });
+  const { pageNumber } = paginationProps;
+
+  const { data: userReviews, status: userReviewsStatus } = useQuery({
+    queryKey: ['userReviews', params.userId, pageNumber],
+    queryFn: () =>
+      getUserReviews({
+        ...paginationProps,
+        userId: params.userId!,
+      }),
+    keepPreviousData: true,
+  });
+
+  const [currentTabIndex, setCurrentTabIndex] = useState(0);
+
+  const handleTabChange = (_e: SyntheticEvent, tabIndex: number) => {
+    setCurrentTabIndex(tabIndex);
+  };
+
+  return (
+    <Box>
+      {status == 'loading' && <LoadingTypography></LoadingTypography>}
+      {status == 'success' && (
+        <>
+          <Paper
+            sx={{
+              p: 2,
+              flexGrow: 1,
+              backgroundColor: (theme) => (theme.palette.mode === 'dark' ? '#1A2027' : '#fff'),
+            }}>
+            <Grid container spacing={2}>
+              <Grid item>
+                <Avatar
+                  alt={data.userName}
+                  src={imgUrl(data.userImageUrl, '/public/autor-szablon.jpg')}
+                  sx={{ width: 200, height: 200, margin: 1 }}
+                />
+              </Grid>
+              <Grid item xs={12} sm container>
+                <Grid item xs container direction="column" spacing={2}>
+                  <Grid item xs>
+                    <Typography gutterBottom variant="h4" component="div">
+                      {data.userName}
+                    </Typography>
+                    <Typography variant="h6" gutterBottom>
+                      <PlaceIcon></PlaceIcon>
+                      {data.userLocation == null ? 'Brak adresu' : data.userLocation}
+                    </Typography>
+                    <Typography variant="body1">Przeczytane książki: {data.readBooksCount}</Typography>
+                  </Grid>
+                </Grid>
+              </Grid>
+            </Grid>
+            <Tabs value={currentTabIndex} onChange={handleTabChange} centered>
+              <Tab label="O mnie" />
+              <Tab label="Ostatnio przeczytane" />
+              <Tab label="Recenzje" />
+            </Tabs>
+            {currentTabIndex === 0 && (
+              <Box sx={{ p: 3 }}>
+                <>
+                  {data.aboutMe != null && data.aboutMe != '' && (
+                    <div>
+                      <Typography variant="h5" gutterBottom>
+                        Kilka słów o mnie
+                      </Typography>
+                      <Typography>{data.aboutMe}</Typography>
+                    </div>
+                  )}
+                  {data.aboutMe == null ||
+                    (data.aboutMe === '' && (
+                      <Typography variant="h5" textAlign={'center'}>
+                        {data.userName} nie dodał swojego opisu
+                      </Typography>
+                    ))}
```

Comment:
Czy coś takiego nie byłoby bardziej czytelne?
```jsx
{!data.aboutMe ? (
    <div>
      <Typography variant="h5" gutterBottom>
        Kilka słów o mnie
      </Typography>
      <Typography>{data.aboutMe}</Typography>
    </div>
  ) : (
      <Typography variant="h5" textAlign={'center'}>
        {data.userName} nie dodał swojego opisu
      </Typography>
)}
```

Reply:
Fakt, chociaż zrobiłeś na odwrót.
A to porównanie z pustym stringiem dlatego, że zrobiłem tak, że jak się nie wpisze nic w formularzu 'O mnie' to nie będzie null, tylko pusty string. Jak był null to klasycznie coś się pierdoliło z zodem i controlled input.